src/bsp/platform/cc3200/xi_bsp_io_fs_cc3200.c

```diff
@@ -0,0 +1,213 @@
+                              const xi_fs_open_flags_t open_flags,
+                              xi_fs_resource_handle_t* resource_handle_out )
+{
+    ( void )open_flags;
+
+#ifdef XI_DEBUG__FOR_FIRMWARE_UPDATE_TESTING_PURPOSES
+    _ReadBootInfo( &sBootInfo );
+#endif
+
+    if ( 1 == xi_bsp_fwu_is_firmware( resource_name ) )
+    {
+        /* the resource is firmware, handle with FLC (flc_api.h) */
+
+        if ( 0 != sl_extlib_FlcOpenFile( "/sys/mcuimgA.bin", size, NULL,
+                                         &firmware_file_handle_last_opened,
+                                         FS_MODE_OPEN_WRITE ) )
+        {
+            firmware_file_handle_last_opened = 0;
+            *resource_handle_out             = 0;
+            return XI_FS_OPEN_ERROR;
+        }
+
+        *resource_handle_out = firmware_file_handle_last_opened;
+    }
+    else
+    {
+        /* it's an ordinary file, handle with file io (fs.h) */
+
+        _i32 file_handle = 0;
+
+        if ( 0 != sl_FsOpen( ( _u8* )resource_name,
+                             xi_bsp_io_fs_open_flags_to_sl_flags( size, open_flags ),
+                             NULL, &file_handle ) )
+        {
+            *resource_handle_out = 0;
+            return XI_FS_OPEN_ERROR;
+        }
+
+        *resource_handle_out = file_handle;
+    }
+
+    return XI_STATE_OK;
+}
+
+#define XI_BSP_IO_FS_READ_BUFFER_SIZE 1024
+
+xi_state_t xi_bsp_io_fs_read( const xi_fs_resource_handle_t resource_handle,
+                              const size_t offset,
+                              const uint8_t** buffer,
+                              size_t* const buffer_size )
+{
+    _i32 result_file_read = 0;
+
+    static _u8 read_buffer[XI_BSP_IO_FS_READ_BUFFER_SIZE];
+
+    if ( resource_handle == firmware_file_handle_last_opened )
+    {
+        result_file_read = sl_extlib_FlcReadFile( resource_handle, offset, read_buffer,
+                                                  XI_BSP_IO_FS_READ_BUFFER_SIZE );
+    }
+    else
+    {
+        result_file_read = sl_FsRead( resource_handle, offset, read_buffer,
+                                      XI_BSP_IO_FS_READ_BUFFER_SIZE );
+    }
+
+    if ( result_file_read < 0 )
+    {
+        return XI_FS_READ_ERROR;
+    }
+    else
+    {
+        *buffer      = read_buffer;
+        *buffer_size = result_file_read;
+
+        return XI_STATE_OK;
+    }
+}
+
+xi_state_t xi_bsp_io_fs_write( const xi_fs_resource_handle_t resource_handle,
+                               const uint8_t* const buffer,
+                               const size_t buffer_size,
+                               const size_t offset,
+                               size_t* const bytes_written )
+{
+    { /* led indicator of file writes */
+#ifdef XI_DEBUG__FOR_FIRMWARE_UPDATE_TESTING_PURPOSES
+        if ( IMG_ACT_USER1 == sBootInfo.ucActiveImg )
+        {
+            GPIO_IF_LedOff( MCU_ORANGE_LED_GPIO );
+            GPIO_IF_LedToggle( MCU_GREEN_LED_GPIO );
```

Comment:
This code / feedback does not touch RED LED at all. This blinks either the green or the orange during file writes depending on the current FW's file position (1 or 2).